src/sbvr-api/odata-response.ts

```diff
@@ -0,0 +1,207 @@
+declare module '@resin/abstract-sql-compiler' {
+	interface AbstractSqlTable {
+		fetchProcessingFields?: {
+			[field: string]: (field: any) => Promise<any>;
+		};
+		localFields?: {
+			[odataName: string]: true;
+		};
+	}
+}
+
+import * as _ from 'lodash';
+import * as Promise from 'bluebird';
+import { Row, Result } from '../database-layer/db';
+import { resolveNavigationResource, resolveSynonym } from './sbvr-utils';
+import { sqlNameToODataName } from '@resin/odata-to-abstract-sql';
+import * as sbvrTypes from '@resin/sbvr-types';
+import {
+	AbstractSqlModel,
+	AbstractSqlTable,
+} from '@resin/abstract-sql-compiler';
+
+const checkForExpansion = Promise.method(
+	(
+		vocab: string,
+		abstractSqlModel: AbstractSqlModel,
+		parentResourceName: string,
+		fieldName: string,
+		instance: Row,
+	) => {
+		let field = instance[fieldName];
+		if (field == null) {
+			return;
+		}
+
+		if (_.isString(field)) {
+			try {
+				field = JSON.parse(field);
+			} catch (_e) {
+				// If we can't JSON.parse the field then we use it directly.
+			}
+		}
+
+		if (_.isArray(field)) {
+			const mappingResourceName = resolveNavigationResource(
+				{
+					abstractSqlModel,
+					vocabulary: vocab,
+					resourceName: parentResourceName,
+				},
+				fieldName,
+			);
+			return process(vocab, abstractSqlModel, mappingResourceName, field).then(
+				expandedField => {
+					instance[fieldName] = expandedField;
+				},
+			);
+		} else {
+			const mappingResourceName = resolveNavigationResource(
+				{
+					abstractSqlModel,
+					vocabulary: vocab,
+					resourceName: parentResourceName,
+				},
+				fieldName,
+			);
+			instance[fieldName] = {
+				__deferred: {
+					uri: '/' + vocab + '/' + mappingResourceName + '(' + field + ')',
+				},
+				__id: field,
+			};
+		}
+	},
+);
+
+export const resourceURI = (
+	vocab: string,
+	resourceName: string,
+	id: string | number,
+) => {
+	if (_.isString(id)) {
+		id = "'" + encodeURIComponent(id) + "'";
+	}
+	return '/' + vocab + '/' + resourceName + '(' + id + ')';
+};
+
+const getLocalFields = (table: AbstractSqlTable) => {
+	if (table.localFields == null) {
+		table.localFields = {};
+		for (const { fieldName, dataType } of table.fields) {
+			if (dataType !== 'ForeignKey') {
+				const odataName = sqlNameToODataName(fieldName);
+				table.localFields[odataName] = true;
+			}
+		}
+	}
+	return table.localFields;
+};
+const getFetchProcessingFields = (table: AbstractSqlTable) => {
+	if (table.fetchProcessingFields == null) {
+		table.fetchProcessingFields = _(table.fields)
+			.filter(
+				({ dataType }) =>
```

Comment:
Prettier becomes weirder and weirder each release -- what sort of formatting is that :/